docs/library.md

```diff
@@ -0,0 +1,237 @@
+member, err = leaderboards.GetMember(context.Background(), leaderboardID, playerID, order, includeTTL)
+if err != nil {
+    return err
+}
+
+playerPrinter(member.PublicID, member.Score, member.Rank)
+```
+
+## Setting and getting member scores in bulk
+
+```
+const leaderboardID = "lbID"
+
+players := leaderboard.Members{
+    &leaderboard.Member{Score: 1000, PublicID: "playerA"},
+    &leaderboard.Member{Score: 2000, PublicID: "playerB"},
+}
+
+err := leaderboards.SetMembersScore(context.Background(), leaderboardID, players, false, "")
+if err != nil {
+    return err
+}
+
+const order = "desc"     //if set to asc, will treat the ranking with ascending scores (less is best)
+const includeTTL = false //if set to true, will return the member's score expiration unix timestamp
+members, err := leaderboards.GetMembers(context.Background(), leaderboardID, []string{"playerA", "playerB"}, order, includeTTL)
+
+for _, member := range members {
+    fmt.Printf("Player(id: %s, score: %d rank: %d)\n", member.PublicID, member.Score, member.Rank)
+}
+```
+
+## Retrieving leaderboard leaders
+
+```
+const leaderboardID = "myleaderboardID"
+
+players := leaderboard.Members{
+    &leaderboard.Member{Score: 10, PublicID: "player1"},
+    &leaderboard.Member{Score: 20, PublicID: "player2"},
+}
+
+err = leaderboards.SetMembersScore(context.Background(), leaderboardID, players, false, "")
+if err != nil {
+    log.Fatalf("leaderboards.SetMembersScore failed: %v", err)
+}
+const pageSize = 10
+const pageIdx = 1 //starts at 1
+leaders, err := leaderboards.GetLeaders(context.Background(), leaderboardID, pageSize, pageIdx, "desc")
+if err != nil {
+    log.Fatalf("leaderboards.GetLeaders failed: %v", err)
+}
+
+for _, player := range leaders {
+    fmt.Printf("Player(id: %s, score: %d rank: %d)\n", player.PublicID, player.Score, player.Rank)
+}
+```
+
+## Incrementing player scores
+
+This code example assumes that there is a playerA score stored on leaderboard lbID
+and  
+```
+const leaderboardID = "lbID"
+const playerID = "playerA"
+const scoreIncrement = 500
+const scoreTTL = ""
+member, err := leaderboards.IncrementMemberScore(context.Background(), leaderboardID, playerID, scoreIncrement,
+    scoreTTL)
+if err != nil {
+    return err
+}
+
+fmt.Printf("Player(id: %s, score: %d rank: %d)\n", member.PublicID, member.Score, member.Rank)
+```
+
+## Number of players on a leaderboard
+
+```
+const leaderboardID = "lbID"
+count, err := leaderboards.TotalMembers(context.Background(), leaderboardID)
+if err != nil {
+    return err
+}
+fmt.Printf("Total number of players on leaderboard %s: %d\n", leaderboardID, count)
+```
+
+## Removing players from a leaderboard
+
+```
+const leaderboardID = "lbID"
+const playerIdToRemove = "playerID"
+err := leaderboards.RemoveMember(context.Background(), leaderboardID, playerIdToRemove) //removing a single player
+if err != nil {
+    return err
+}
+
+playerIDsToRemove := make([]interface{}, 2)
```

Comment:
Not related to this change, but shouldn't we receive an array of strings instead of an array of interfaces?

Reply:
This is because of the internal Redis lib API:
ZRem(key string, members ...interface{}) *redis.IntCmd

Reply:
got it